### PR TITLE
Pluggable components

### PR DESCRIPTION
**Pull request by tiberiuichim**
WIP

### CODE REVIEW

**Comment by sneridagh** [comment]
Love it...

**Review by sneridagh — APPROVED** [review]

**Comment by sneridagh** [comment]
Some narrative docs would be great!

**Comment by tiberiuichim** [comment]
@sneridagh I'll add narative docs.

Should we merge the slots with the "pluggables" implementation? It would simplify the story we tell.

The difference between slots and pluggables right now is:

- slots are more declarative, the "inserts" are declared as configuration (or they come from the content)
- the "pluggables" are more dynamic and short-lived in their nature, similar to the Portal components that we use. The pluggables might also be used as an extension mechanism that's not possible with slots. Each pluggable slot can process the "inserts" in any way it wants (not just rendering them), so they can be used as an adhoc mini-plugin framework.

**Review by sneridagh — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     }
   ],
   "license": "MIT",
-  "version": "12.10.1",
+  "version": "12.8.0",
```
Comment (sneridagh) [inline package.json:12]: Watch out!

**Pull request merged**